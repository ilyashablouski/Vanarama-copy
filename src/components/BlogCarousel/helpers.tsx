import { ProductDerivativeFilter } from '../../../generated/globalTypes';
import { BlogPostCarouselData_blogPost_productFilter } from '../../../generated/BlogPostCarouselData';

const vehicleTypeMapper = {
  CAR: ['Car'],
  LCV: ['Van', 'Pickup'],
};
// eslint-disable-next-line import/prefer-default-export
export const productFilterMapper = (
  blogProductFilter?: Nullable<BlogPostCarouselData_blogPost_productFilter>,
): ProductDerivativeFilter => ({
  manufacturerNames: blogProductFilter?.manufacturer || undefined,
  rangeNames: blogProductFilter?.range || undefined,
  bodyStyles: blogProductFilter?.bodyType || undefined,
  fuelTypes: blogProductFilter?.fuelType || undefined,
  transmissions: blogProductFilter?.transmission || undefined,
  vehicleCategory:
    vehicleTypeMapper[
      blogProductFilter?.vehicleType?.[0] as keyof typeof vehicleTypeMapper
    ] || undefined,
});
