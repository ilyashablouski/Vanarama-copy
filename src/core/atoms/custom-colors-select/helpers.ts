import { SelectOptionList } from 'core/atoms/custom-colors-select/interface';

function getFilteredList(optionData: (SelectOptionList | null)[]) {
  return optionData?.reduce<(SelectOptionList | null)[][]>((acc, outerItem) => {
    if (!acc.length) {
      acc.push([outerItem]);

      return acc;
    }

    const foundedIndex = acc.findIndex(item => {
      if (!item[0]) {
        return -1;
      }
      return outerItem?.leadTime === item[0].leadTime;
    });

    if (foundedIndex >= 0) {
      acc[foundedIndex].push(outerItem);
    } else {
      acc.push([outerItem]);
    }

    return acc;
  }, []);
}

function getStructuredList(
  optionList: (SelectOptionList | null)[] | null | undefined,
) {
  if (!optionList) {
    return null;
  }

  const onOfferItems = optionList.filter(item => item?.onOffer);
  const nonOfferItems = optionList.filter(item => !item?.onOffer);

  const filteredOnOfferItems = getFilteredList(onOfferItems);
  const filteredNonOfferItems = getFilteredList(nonOfferItems);

  return [...filteredOnOfferItems, ...filteredNonOfferItems];
}

export default getStructuredList;
