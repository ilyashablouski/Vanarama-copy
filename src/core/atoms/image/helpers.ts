import { ImageLoaderProps } from 'next/image';
import { trimStartSlash } from '../../../utils/url';

// eslint-disable-next-line import/prefer-default-export
export function cloudflareLoader({ src, width, quality }: ImageLoaderProps) {
  const params = [`width=${width}`];

  if (quality) {
    params.push(`quality=${quality}`);
  }

  const paramsString = params.join(',');

  return `${
    process.env.IMG_OPTIMISATION_HOST
  }/cdn-cgi/image/${paramsString}/${trimStartSlash(src)}`;
}
