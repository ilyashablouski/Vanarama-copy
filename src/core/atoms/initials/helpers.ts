const getInitials = (fullName: string) =>
  fullName
    .match(/(^\S\S?|\b\S)?/g)
    ?.join('')
    .match(/(^\S|\S$)?/g)
    ?.join('')
    .toUpperCase();

export default getInitials;
