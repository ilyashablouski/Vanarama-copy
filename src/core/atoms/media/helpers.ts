// eslint-disable-next-line import/prefer-default-export
export const getYouTubeThumbnail = (id: string) =>
  `https://img.youtube.com/vi_webp/${id}/sddefault.webp`;
