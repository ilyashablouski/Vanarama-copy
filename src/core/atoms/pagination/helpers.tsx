// eslint-disable-next-line import/prefer-default-export
export function addEtc(pagesArray: number[]): string[] {
  const stringPagesArray = pagesArray.map(page => page.toString());
  if (pagesArray.length <= 1) {
    return stringPagesArray;
  }
  const lastPage = pagesArray[pagesArray.length - 1];
  const secondLast = pagesArray[pagesArray.length - 2];
  if (lastPage !== secondLast + 1) {
    stringPagesArray.splice(pagesArray.length - 1, 0, '...');
  }
  if (pagesArray[0] + 1 !== pagesArray[1]) {
    stringPagesArray.splice(1, 0, '...');
  }
  return stringPagesArray;
}
