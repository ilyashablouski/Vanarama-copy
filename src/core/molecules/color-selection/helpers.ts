export const baseClassName = 'color-selection';

export function getClassName(elementClassName: string) {
  return `${baseClassName}__${elementClassName}`;
}
