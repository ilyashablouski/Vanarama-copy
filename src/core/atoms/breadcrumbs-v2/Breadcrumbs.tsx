import React, { FC, memo } from 'react';
import BreadcrumbsItem from './BreabcrumbsItem';
import { IBreadcrumbLink, IBreadcrumbProps } from './helpers';

const Breadcrumbs: FC<IBreadcrumbProps> = memo(({ items, dataTestId }) => {
  if (!items?.length) {
    return null;
  }

  const renderBreadcrumbsItem = (itemData: IBreadcrumbLink, index: number) => {
    // Back breadcrumb link for mobile version
    const isMobileBackBreadcrumbLink = items.length - 2 === index;
    const isLastBreadcrumbsItem = items.length === index + 1;

    return (
      <BreadcrumbsItem
        isLastItem={isLastBreadcrumbsItem}
        isMobileBackLink={isMobileBackBreadcrumbLink}
        itemData={itemData}
      />
    );
  };

  return (
    <nav data-testid={dataTestId ?? 'breadcrumbs'}>
      <ul className="breadcrumb">{items.map(renderBreadcrumbsItem)}</ul>
    </nav>
  );
});

export default Breadcrumbs;
