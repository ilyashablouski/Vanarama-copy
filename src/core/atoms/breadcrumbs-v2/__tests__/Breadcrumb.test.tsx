import React from 'react';
import renderer from 'react-test-renderer';

import Breadcrumbs from '../Breadcrumbs';
import { IBreadcrumbProps } from '../helpers';

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
  })),
});

const optionalProps = {
  dataTestId: 'breadcrumbs',
  items: [
    { link: { href: '/', label: 'Page1' } },
    { link: { href: '/', label: 'Page2' } },
    { link: { href: '/', label: 'Page3' } },
  ],
};

function getComponent(props?: IBreadcrumbProps) {
  return renderer.create(<Breadcrumbs {...props} />).toJSON();
}

describe('<Breadcrumbs />', () => {
  it('renders correctly with optional props', () => {
    const tree = getComponent(optionalProps);
    expect(tree).toMatchSnapshot();
  });
});
