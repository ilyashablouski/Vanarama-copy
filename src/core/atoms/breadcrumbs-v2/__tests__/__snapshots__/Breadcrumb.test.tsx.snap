// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumbs /> renders correctly with optional props 1`] = `
Array [
  <nav
    className="breadcrumbs breadcrumbs--mobile"
    data-testid="breadcrumbs"
  >
    <ul
      className="breadcrumbs__list"
    >
      <li
        className="breadcrumbs__item"
      >
        <a
          className="breadcrumbs__link link -teal -small"
          data-menu={null}
          data-testid="router-link"
          href="/"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <i
            className="icon -teal"
          >
            <svg
              height="1em"
              viewBox="0 0 512 512"
              width="1em"
            >
              <polyline
                points="328 112 184 256 328 400"
                style={
                  Object {
                    "fill": "none",
                    "stroke": "var(--class-color)",
                    "strokeLinecap": "round",
                    "strokeLinejoin": "round",
                    "strokeWidth": 48,
                  }
                }
              />
            </svg>
          </i>
          Back to 
          Page2
        </a>
      </li>
    </ul>
  </nav>,
  <nav
    className="breadcrumbs breadcrumbs--desktop"
    data-testid="breadcrumbs"
  >
    <ul
      className="breadcrumbs__list"
    >
      <li
        className="breadcrumbs__item"
      >
        <a
          className="breadcrumbs__link link -teal -small"
          data-menu={null}
          data-testid="router-link"
          href="/"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          Page1
        </a>
        <div
          style={
            Object {
              "color": "#ccc",
              "fontSize": "24px",
              "lineHeight": 1.75,
              "opacity": 0.75,
            }
          }
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </div>
      </li>
      <li
        className="breadcrumbs__item"
      >
        <a
          className="breadcrumbs__link link -teal -small"
          data-menu={null}
          data-testid="router-link"
          href="/"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          Page2
        </a>
        <div
          style={
            Object {
              "color": "#ccc",
              "fontSize": "24px",
              "lineHeight": 1.75,
              "opacity": 0.75,
            }
          }
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </div>
      </li>
      <li
        className="breadcrumbs__item"
      >
        <div
          style={
            Object {
              "color": "#ccc",
              "fontSize": "24px",
              "lineHeight": 1.75,
              "opacity": 0.75,
            }
          }
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </div>
      </li>
    </ul>
  </nav>,
]
`;
